sysdeps/managarm/generic/ioctl.cpp

```diff
@@ -323,6 +324,8 @@ int sys_ioctl(int fd, unsigned long request, void *arg, int *result) {
 		if(send_req.error())
 			return EINVAL;
 		HEL_CHECK(send_req.error());
+		if(imbue_creds.error())
+			return EINVAL;
```

Comment:
Is this right? Don't really know all that well how our permissions system (or the lack thereof...) works, but if this would fail if one does not have sufficient permissions should it fail with some different error?

Reply:
IIRC this is getting tripped if an error/dismiss is returned instead of Credentials.

Reply:
Shouldn't this be a fatal error? I don't understand why we'd return EINVAL on ``send_req`` or ``imbue_creds`` failing to send.